test/gtest/wat-lexer.cpp

```diff
@@ -374,6 +376,361 @@ TEST(LexerTest, LexHexInt) {
   }
 }
 
+TEST(LexerTest, ClassifyInt) {
+  {
+    Lexer lexer("0"sv);
+    ASSERT_NE(lexer, lexer.end());
+
+    ASSERT_TRUE(lexer->getU64());
+    ASSERT_TRUE(lexer->getS64());
+    ASSERT_TRUE(lexer->getI64());
+    ASSERT_TRUE(lexer->getU32());
+    ASSERT_TRUE(lexer->getS32());
+    ASSERT_TRUE(lexer->getI32());
+    ASSERT_TRUE(lexer->getF64());
+    ASSERT_TRUE(lexer->getF32());
+
+    EXPECT_EQ(*lexer->getU64(), 0ull);
+    EXPECT_EQ(*lexer->getS64(), 0ll);
+    EXPECT_EQ(*lexer->getI64(), 0ull);
+    EXPECT_EQ(*lexer->getU32(), 0u);
+    EXPECT_EQ(*lexer->getS32(), 0);
+    EXPECT_EQ(*lexer->getI32(), 0u);
+    EXPECT_EQ(*lexer->getF64(), 0.0);
+    EXPECT_EQ(*lexer->getF32(), 0.0);
+    EXPECT_FALSE(std::signbit(*lexer->getF64()));
+    EXPECT_FALSE(std::signbit(*lexer->getF32()));
+  }
+  {
+    Lexer lexer("+0"sv);
+    ASSERT_NE(lexer, lexer.end());
+
+    EXPECT_FALSE(lexer->getU64());
+    ASSERT_TRUE(lexer->getS64());
```

Comment:
Whey do you use `ASSERT` and when to use `EXPECT`?

Reply:
If an `ASSERT` fails, it prevents the rest of the test function from executing, so it's used whenever later statements in the test only make sense if the assertion does not fail. `EXPECT`, on the other hand, continues executing the test function even if it fails, so it can produce more helpful error messages and should be used wherever possible.